src/cql/upstream-external/CirgLibraryQuestionnaireLogic_DCW.cql

```diff
@@ -0,0 +1,301 @@
+  (singleton from (((responses) o where o.linkId.value = linkId) I
+  return 
+    if I is null then ({null as Integer})
+    else
+      if I.answer[0].value.code is not null then getAnswerValueByCode(questionnaire, I.answer[0].value.code.value) else 0 as Integer))
+
+/*
+ * return questionnaire responses of type FHIR.QuestionnaireResponse matched to the target questionnaire
+ * @param responses as FHIR.QuestionnaireResponse
+ * @param questionnaire as FHIR.Questionnaire to match against
+ */
+define function MatchedReponsesByQuestionnaire(responses List<FHIR.QuestionnaireResponse>, targetQuestionnaire FHIR.Questionnaire):
+  ((responses) Q
+    where 
+    (
+      (
+        PositionOf(Lower(targetQuestionnaire.name), Lower(Q.questionnaire.value)) != -1 or
+        PositionOf(Lower(targetQuestionnaire.id), Lower(Q.questionnaire.value)) != -1
+      ) and 
+      Q.status.value = 'completed')
+    sort by authored desc, meta.lastUpdated.value desc
+  )
+  
+
+/*
+ * return answer value FHIR object by response item of FHIR.QuestionnaireResponse.Item
+ * @param linkId of the response item as String
+ * @param responses as a list of FHIR.QuestionnaireResponse.Item
+ */
+define function GetResponseValueByLinkId(linkId String, responses List<FHIR.QuestionnaireResponse.Item>):
+  (responses r where r.linkId.value = linkId) I
+  return I.answer[0].value
+
+/*
+ * return FHIR Questionnaire object
+ * 
+ */
+define function getQuestionnaire(matchedString String):
+  First([Questionnaire] Q
+    where 
+      PositionOf(Lower(matchedString), Lower(Q.id)) != -1 or
+      PositionOf(Lower(matchedString), Lower(Q.name)) != -1 or
+      PositionOf(Lower(Q.name), Lower(matchedString)) != -1 or
+      Lower(Q.name) = Lower(matchedString)
+  )
+
+// Logic for presenting questionnaire
+
+// default scheduled timing for administering questionnaire
+define Scheduled_Timing: Tuple {
+  repeat: {
+    frequency: 1,
+    period: 1,
+    periodUnit: 'd'
+  }
+}
+
+define dcw_care_partner_present:
+// TODO check Care Partner present (via CP gateway question, or field in FHIR?)
+  true
+
+define dcw_cog_level_1:
+// if (no dementia condition)
+  not HasDementia
+
+define dcw_cog_level_2:
+// if (dementia condition)
+// and/or 
+// CIRG-MINICOG score is fail
+// a failed score is 0,1,2, or 3
+  if HasDementia is true then true
+  else if First(MINICOG_Score) <= 3 then true
+  else false
+
+define dcw_mood_level_1:
+  dcw_care_partner_present
+
+define dcw_mood_level_2:
+// if (
+//   depression condition ("F32.A - Depression, unspecified" and "F33.1 - Major Depressive Disorder, Recurrent, Moderate")
+//   and/or 
+//   anxiety condition ("F41.1 - Generalized Anxiety Disorder" - not present in our 3 test patients...)
+//   and/or 
+//   CIRG-BEHAV5 score is fail 
+// )
+  if HasMoodConditions is true then true 
+  else if First(BEHAV5_Score) <= 7 then true
```

Comment:
@sierramatice Is a BEHAV5 score <= 7 a failed score? If not, what is considered a failed BEHAV5 score?

Reply:
A failed BEHAV5 score is 1 or higher. (If yes is answered to any questions) and the open response field is not factored in